### ISSUE TITLE
`–files-from` does not work when `copy/move` with `–drive-shared-with-me` tag.

### ISSUE THREAD

**Issue by geart891** [rclone/rclone#3619]
#### What is the problem you are having with rclone?
`copy/move` commands do not work with `--files-from` and `--drive-shared-with-me` tag.

#### What is your rclone version (output from `rclone version`)
1.49.5
#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
The problem is apparently is on all OS, (Ubuntu, Windows 10 64bit, etc)

####  Which cloud storage system are you using? (eg Google Drive)
Google Drive.

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
`rclone copy remote1: /home/content --drive-shared-with-me --files-from=upload.txt --stats-one-line -vv`

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

```
# upload.txt
Prison.Break.S05E01.Ogygia.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E02.Kaniel.Outis.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E03.The.Liar.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E04.The.Prisoners.Dilemma.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E05.Contingency.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E06.Phaeacia.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E07.Wine-Dark.Sea.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E08.Progeny.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
```
Logs from `lsf` command:
`rclone lsf remote1: --drive-shared-with-me --files-from upload.txt`
```
$ rclone lsf mnc: --drive-shared-with-me --files-from upload.txt
Prison.Break.S05E01.Ogygia.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E02.Kaniel.Outis.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E03.The.Liar.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E04.The.Prisoners.Dilemma.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E05.Contingency.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E06.Phaeacia.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E07.Wine-Dark.Sea.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E08.Progeny.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
```
```
$ rclone lsf remote1: --drive-shared-with-me --files-from upload.txt
Prison.Break.S05E01.Ogygia.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E02.Kaniel.Outis.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E03.The.Liar.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E04.The.Prisoners.Dilemma.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E05.Contingency.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E06.Phaeacia.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E07.Wine-Dark.Sea.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
Prison.Break.S05E08.Progeny.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-EPSiLON.mkv
```
Logs from `copy` command:
`rclone copy remote1: --drive-shared-with-me --files-from upload.txt /root/content/ --dry-run -vv --stats-one-line`
```
$ rclone copy mnc: --drive-shared-with-me --files-from upload.txt /root/content/ --dry-run -vv --stats-one-line
2019/10/10 14:47:40 DEBUG : rclone: Version "v1.49.5" starting with parameters ["F:\\OneDrive\\System\\rClone\\rclone.exe" "copy" "remote1:" "--drive-shared-with-me" "--files-from" "upload.txt" "/root/content/" "--dry-run" "-vv" "--stats-one-line"]
2019/10/10 14:47:40 DEBUG : Using config file from "C:\\Users\\Rhaps\\.config\\rclone\\rclone.conf"
2019/10/10 14:47:40 DEBUG : Bloodstained Ritual of the Night.lnk: Excluded
2019/10/10 14:47:40 DEBUG : CCleaner.lnk: Excluded
2019/10/10 14:47:40 DEBUG : class yenhoa: Excluded
2019/10/10 14:47:40 DEBUG : command.txt: Excluded
2019/10/10 14:47:40 DEBUG : CPUZ.lnk: Excluded
2019/10/10 14:47:40 DEBUG : desktop.ini: Excluded
2019/10/10 14:47:40 DEBUG : EVKey.lnk: Excluded
2019/10/10 14:47:40 DEBUG : Examples: Excluded
2019/10/10 14:47:40 DEBUG : FINAL FANTASY VIII REMASTERED.lnk: Excluded
2019/10/10 14:47:40 DEBUG : foobar2000.lnk: Excluded
2019/10/10 14:47:40 DEBUG : Git.lnk: Excluded
2019/10/10 14:47:40 DEBUG : GPUZ.lnk: Excluded
2019/10/10 14:47:40 DEBUG : HWiNFO64.lnk: Excluded
2019/10/10 14:47:40 DEBUG : Internet Download Manager.lnk: Excluded
2019/10/10 14:47:40 DEBUG : Irfanview.lnk: Excluded
2019/10/10 14:47:40 DEBUG : LOGGING.CSV: Excluded
2019/10/10 14:47:40 DEBUG : MPV.lnk: Excluded
2019/10/10 14:47:40 DEBUG : Mqueue.txt: Excluded
2019/10/10 14:47:40 DEBUG : New Electro & House 2012 Dance Mix #17.aac: Excluded
2019/10/10 14:47:40 DEBUG : New folder: Excluded
2019/10/10 14:47:40 DEBUG : qBittorrent.lnk: Excluded
2019/10/10 14:47:40 DEBUG : RAM.png: Excluded
2019/10/10 14:47:40 DEBUG : Rclone Browser.lnk: Excluded
2019/10/10 14:47:40 DEBUG : Steam.lnk: Excluded
2019/10/10 14:47:40 DEBUG : The Witcher III.lnk: Excluded
2019/10/10 14:47:40 DEBUG : Turn off Monitor.lnk: Excluded
2019/10/10 14:47:40 DEBUG : UFMagnet.txt: Excluded
2019/10/10 14:47:40 DEBUG : upload.txt: Excluded
2019/10/10 14:47:40 DEBUG : Visual Studio Code.lnk: Excluded
2019/10/10 14:47:40 DEBUG : XLaunch.lnk: Excluded
2019/10/10 14:47:40 DEBUG : Xming.lnk: Excluded
2019/10/10 14:47:40 DEBUG : Yamaha Fes: Excluded
2019/10/10 14:47:43 DEBUG : Copy: Excluded from sync (and deletion)
2019/10/10 14:47:43 INFO  : Local file system at \\?\F:\OneDrive\DSKDesktop: Waiting for checks to finish
2019/10/10 14:47:43 INFO  : Local file system at \\?\F:\OneDrive\DSKDesktop: Waiting for transfers to finish
2019/10/10 14:47:43 INFO  :          0 / 0 Bytes, -, 0 Bytes/s, ETA -
2019/10/10 14:47:43 DEBUG : 11 go routines active
2019/10/10 14:47:43 DEBUG : rclone: Version "v1.49.5" finishing with parameters ["F:\\OneDrive\\System\\rClone\\rclone.exe" "copy" "remote1:" "--drive-shared-with-me" "--files-from" "upload.txt" "/root/content/" "--dry-run" "-vv" "--stats-one-line"]
```

I have report this problem in the rclone forum. Link: https://forum.rclone.org/t/files-from-does-not-work-when-copy-in-drive-shared-with-me/12197/4

**Comment by ncw** [comment]
Hmm, this is a bug.

You can work-around with `--no-traverse`.

I bisected this problem down to this commit

https://github.com/rclone/rclone/commit/a28239f005aeb290528e8373d5b99d2efa34c870

**Comment by ncw** [comment]
Can you have a go with this fix please?

I found a pretty major problem with --files-from and I think hopefully it should address your issue.

https://beta.rclone.org/branch/v1.49.5-215-gcb873973-fix-3619-no-traverse-beta/ (uploaded in 15-30 mins)

**Comment by geart891** [comment]
Thank for your quick response. It works as intended in the build in your post (ie. without the `--no-traverse`). Tested both on windows and linux.

**Comment by ncw** [comment]
Thank you for testing :-)

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.50
